src/components/UserPlan.tsx

```diff
@@ -1,42 +1,207 @@
+              <BuyContainer>
+                <p>
+                  {Number(maxSubscribers) - Number(remainingCredits)}{" "}
+                  {isPlanTypeSuscribers ? (
+                    <FormattedMessage id="header.plan_suscribers" />
+                  ) : (
+                    <FormattedMessage id="header.plan_emails" />
+                  )}{" "}
+                  (<strong>{remainingCredits}</strong>{" "}
+                  <FormattedMessage id="header.availables" />)
+                </p>
+              </BuyContainer>
+            ) : (
+              <BuyContainer>
+                <p>
+                  <strong>{remainingCredits}</strong> {description}
+                </p>
+              </BuyContainer>
+            )}
+            {!!Object.keys(sms).length && smsEnabled && (
+              <BuyContainer>
+                <p>
+                  <strong>
+                    US${" "}
+                    <FormattedNumber
+                      // eslint-disable-next-line react/style-prop-object
+                      style="decimal"
+                      value={smsRemainingCredits}
+                      minimumFractionDigits={2}
+                      maximumFractionDigits={2}
+                    />
+                  </strong>{" "}
+                  {smsDescription}
+                </p>
+                {smsButtonUrl && (
+                  <a className="user-plan" target="_self" href={smsButtonUrl}>
+                    {smsButtonText}
+                  </a>
+                )}
+              </BuyContainer>
+            )}
+          </UserPlanType>
+        </>
+      )}
+
+      {hasClientManager && (
+        <UserPlanType>
+          {clientManager?.profileName && (
+            <MonthlyPlan>
+              <FormattedMessage id="header.profile" />{" "}
+              <strong>{clientManager?.profileName}</strong>
+            </MonthlyPlan>
+          )}
+          {!clientManager?.profileName && (
+            <>
+              <MonthlyPlan>
+                <FormattedMessage id="header.send_mails" />
+              </MonthlyPlan>
+              <p className="user-plan-enabled">
+                <FormattedMessage id="header.enabled" />
+              </p>
+            </>
+          )}
+        </UserPlanType>
+      )}
+      <Modal isOpen={isModalOpen} handleClose={() => setIsModalOpen(false)}>
+        <UpgradePlanForm
+          isSubscriber={isSubscribers}
+          handleClose={() => setIsModalOpen(false)}
+          user={user}
+        />
+      </Modal>
     </div>
   );
 };
+
+const MonthlyPlan = ({ children }: { children: React.ReactNode }) => {
+  return <p className="user-plan--monthly-text">{children}</p>;
+};
+
+const UserPlanType = ({ children }: { children: React.ReactNode }) => (
+  <div className="user-plan--type">{children}</div>
+);
+
+const BuyContainer = ({ children }: { children: React.ReactNode }) => {
+  return <div className="user-plan--buyContainer">{children}</div>;
+};
```

Comment:
```suggestion
const MonthlyPlan = ({ children }: { children: React.ReactNode }) => {
  return <p className="user-plan--monthly-text">{children}</p>;
};
interface UserPlanTypeProps{
   profileName: string;
}
const UserPlanType = ({ profileName}: UserPlanTypeProps) => (
  <div className="user-plan--type">
     <MonthlyPlan>
        <FormattedMessage id="header.profile" />{" "}
        <strong>{profileName}</strong>
   </MonthlyPlan>
  </div>
);

const BuyContainer = ({ children }: { children: React.ReactNode }) => {
  return <div className="user-plan--buyContainer">{children}</div>;
};
```

Esta parte de acá me parece pensado en átomos pero no se si tiene sentido porque el CSS esta escrito de una forma muy especifica, creo que el _.user-plan--monthly-text_ debe estar dentro de _.user-plan--type_. Si este es el caso recomiendo hacer que el componente <UserPlanType/> reciba un objeto PlanType o reciba los props por separado

Reply:
`UserPlanType`, `BuyContainer` y `MonthlyPlan` basicamente hacen de containers, son un elemento y una clase bien específica. 
MontlyPlan puede ser o no un children de UserPlanType y no siempre se renderiza por las mismas condiciones, por eso al final me pareció mas legible pasarlo como children cuando corresponda (ademas varían los componentes "siblings" que puede tener) 
Creo que hacer  un componente UserPlanType, que contemple todas las opciones, iba a quedar tan complejo como el return de `UserPlan`
El código sugerido si se ve mas sencillo pero no veo las condiciones de las que depende `MonthlyPlan` y tampoco a `BuyContainer`, avisame si te parece hacer una call para conversarlo en detalle
Gracias!